tractor/__init__.py

```diff
@@ -77,21 +77,33 @@ class RemoteActorError(ActorFailure):
             coro = func(**kwargs)
 
             if inspect.isasyncgen(coro):
-                async for item in coro:
-                    # TODO: can we send values back in here?
-                    # it's gonna require a `while True:` and
-                    # some non-blocking way to retrieve new `asend()`
-                    # values from the channel:
-                    # to_send = await chan.recv_nowait()
-                    # if to_send is not None:
-                    #     to_yield = await coro.asend(to_send)
-                    await chan.send({'yield': item, 'cid': cid})
-
-                log.debug(f"Finished iterating {coro}")
-                # TODO: we should really support a proper
-                # `StopAsyncIteration` system here for returning a final
-                # value if desired
-                await chan.send({'stop': None, 'cid': cid})
+                with trio.open_cancel_scope() as cs:
+                    async for item in coro:
+                        # TODO: can we send values back in here?
```

Comment:
@vodik interestingly enough I had a bug (just below) in this section but this comment is relevant to some of the stuff we talked about. My main outstanding concern here is that I'm not sure how to non-blocking read from the underlying socket if nothing has been `asend()` from the other side (assuming we don't send an actual `None` through on every iteration). I know you said maybe that's how it **should** work but it still seems like unnecessary overhead to me..